apps/navstar_dns/src/navstar_dns_poll_fsm.erl

```diff
@@ -694,6 +729,69 @@ zone_records_fake_test() ->
     {ok, [ExpectedRecords]} = file:consult(RecordFileName),
     ?assertEqual(ExpectedRecords, Records).
 
+zone_records_single_agentip_test() ->
+    DataDir = code:priv_dir(navstar_dns),
+    Filename = filename:join(DataDir, "fake.json"),
+    {ok, Data} = file:read_file(Filename),
+    {ok, ParsedBody} = mesos_state_client:parse_response(Data),
+    Tasks = mesos_state_client:tasks(ParsedBody),
+    [A|_T] = Tasks,
+    Result = navstar_dns_poll_fsm:task_ip_by_agent(A),
+    ExpectedResult = {<<"agentip">>, {1, 2, 3, 12}},
+    ?assertEqual(ExpectedResult, Result).
+
+zone_records_single_autoip_test() ->
+    DataDir = code:priv_dir(navstar_dns),
+    Filename = filename:join(DataDir, "fake.json"),
+    {ok, Data} = file:read_file(Filename),
+    {ok, ParsedBody} = mesos_state_client:parse_response(Data),
+    Tasks = mesos_state_client:tasks(ParsedBody),
+    [A|_T] = Tasks,
+    Result = navstar_dns_poll_fsm:task_ip_autoip(A),
+    ExpectedResult = {<<"autoip">>, {1, 2, 3, 12}},
+    ?assertEqual(ExpectedResult, Result).
+
+zone_records_agentip_test() ->
+    DataDir = code:priv_dir(navstar_dns),
+    Filename = filename:join(DataDir, "fake.json"),
+    {ok, Data} = file:read_file(Filename),
+    {ok, ParsedBody} = mesos_state_client:parse_response(Data),
+    Tasks = mesos_state_client:tasks(ParsedBody),
+    AgentIPList = [navstar_dns_poll_fsm:task_ip_by_agent(X) || X <- Tasks],
+    %% ?debugFmt("Agentip List: ~p", [AgentIPList]).
+    ?assertEqual(expected_agentip_list(), AgentIPList).
+
+expected_agentip_list() ->
+    [{<<"agentip">>, {1, 2, 3, 12}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 12}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}},
+    {<<"agentip">>, {1, 2, 3, 11}}].
+
+zone_records_autoip_test() ->
```

Comment:
These tests are great. Does it make sense to get a state.json file that would trigger the heuristic? I think all it's a matter of doing is launching an application on Docker bridge, docker host, and docker bridge, and adding a port mapping. You should be able to do this with any DC/OS.